jetty9-base/src/main/java/com/google/cloud/runtimes/jetty9/TracingLogHandler.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (C) 2016 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.runtimes.jetty9;
+
+import java.util.logging.ConsoleHandler;
+import java.util.logging.LogRecord;
+
+/**
+ * A Logging Handler extended with a request traceid label.
+ */
+public class TracingLogHandler extends ConsoleHandler {
+
+  /**
+   * Construct a TracingLogHandler.
+   */
+  public TracingLogHandler() {}
+
+  @Override
+  public synchronized void publish(LogRecord record) {
+    String traceid = RequestContextScope.getCurrentTraceid();
+    if (traceid != null) {
+      record.setMessage(record.getMessage() + " traceid=" + traceid);
```

Comment:
You're sort of doing some formatting here by appending the `traceId` to the message. Can we instead add `traceId` as a parameter to `LogRecord.paramters` and output it via the log message formatter.  We may need to create a custom formatter rather than using `java.util.logging.SimpleFormatter`. This also opens the door to potentially having a JSON formatter that can be processed via fluentd into Stackdriver Logging.